app/ide-desktop/lib/client/src/index.ts

```diff
@@ -47,16 +72,57 @@ class App {
     isQuitting = false
 
     async run() {
-        const { args, windowSize, chromeOptions } = configParser.parseArgs()
-        this.args = args
+        // On macOS when Enso-associated file is opened, the application is first started and then it
+        // receives the `open-file` event. However, if there is already an instance of Enso running,
+        // it receives the `open-file` event (and no new instance is created for us). In this case,
+        // we manually start a new instance of the application and pass the file path to it (using the
+        // Windows-style command).
+        electron.app.on('open-file', (event, path) => {
+            const extension = pathModule.extname(path).toLowerCase()
+            if (
+                extension === `.${paths.BUNDLED_PROJECT_EXTENSION}` ||
+                extension === `.${paths.SOURCE_FILE_EXTENSION}`
+            ) {
+                // Not the file extension we care about, let the default handler handle it.
+                return
```

Comment:
1. we are trying not to use early exists, couldd you just use `if` without return, please?
2. I don't understand it. You are checking the extensions and if they are our target extensions we are not handling them? Is the if condition correct? If it is, I need some explanation in the code here.